### ISSUE TITLE
Option to use the group color of person being mentioned

### ISSUE THREAD

**Issue by Blazeflack** [paul999/mention#48]
It would be awesome if you could add an option to use the group color of the person being mentioned when posting the mention.

Right now the default color for a mention is bright red. If I mention "Paul" who is in a group with a blue group color, then I would like the mention in the forum post to be in that specific blue color.

Thanks for your consideration.

**Comment by yannickboy15** [comment]
Agreed, this would be great to have as I got no clue where to do it myself right now.

**Comment by Galixte** [comment]
Hi @paul999,

could you take a look?

**Comment by paul999** [comment]
No, I probarbly won't add this in the near feature.

**Comment by karimcambridge** [comment]
Duplicate of #30

**Comment by paul999** [comment]
Not duplicate at all. They are not even related :). The problem with using the specific group color is that it dynamic, and as such I will need to do much more queries which will make the extension less performant. I haven't found a simple way to do this, and thats why it hasn't been added (And probarbly won't be added soon)

**Comment by HumbleDeer** [comment]
> I will need to do much more queries which will make the extension less performant. I haven't found a simple way to do this...

Given that the extension only requires this information when you're actually mentioning someone, wouldn't the best solution here be to use javascript to callback to the backend, where it is looked up on demand? I can't imagine users performing so abundantly many that this causes large scale issues with performance.

If on the other hand you're interested in changing the colour retroactively as well, then I imagine it's more than fine to just cache them for a reasonable amount of time? And hook into the functions that change a user's colour or the ones that are built-in that relate to the user's colour, to then call your plugin in some way to update its cache. 

Heck, perhaps such an item already exists in cache somewhere. 

Let me know whether this sounds by any means feasable or useful as a method for achieving this. I'm interested in seeing whether this is worth my time trying to implement it myself. :)
